modernizer-maven-plugin/src/test/java/org/gaul/modernizer_maven_plugin/ModernizerSuppressionsTest.java

```diff
@@ -0,0 +1,287 @@
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreConstructorTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreGenericClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreGenericClassConstructorTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodInGenericClassTest;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodReturningArrayClassTest;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodReturningArrayPrimitiveTypeClassTest;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodReturningDeclaredTypeClassTest;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodReturningGenericTypeClassTest;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodReturningPrimitiveTypeClassTest;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodWithArrayTypeParametersTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodWithDeclaredTypeParametersTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodWithEmptyParametersTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodWithGenericTypeParametersTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodWithPrimitiveAndGenericTypeParametersTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodWithPrimitiveTypeParametersTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodWithVoidParameterTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreOverloadedMethodInGenericClassTest;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreOverloadedMethodTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreStaticOuterClassConstructor.IgnoreStaticInnerClassConstructor;
+import org.junit.Before;
+import org.junit.Test;
+import org.objectweb.asm.ClassReader;
+
+public final class ModernizerSuppressionsTest {
+    private Map<String, Violation> violations;
+    private static final Collection<String> NO_EXCLUSIONS =
+        Collections.<String>emptySet();
+    private static final Collection<Pattern> NO_EXCLUSION_PATTERNS =
+        Collections.<Pattern>emptySet();
+    private static final Collection<String> NO_IGNORED_PACKAGES =
+        Collections.<String>emptySet();
+    private static final Collection<String> NO_IGNORED_METHODS =
+        Collections.<String>emptySet();
+    private List<Pattern> ignoreClasses = new ArrayList<Pattern>();
```

Comment:
It would be better to not initialize them. The tests could fail in a confusing way if this is left as an empty list.

Reply:
Do you think the initialization should be done in `setUp()` ?

Reply:
They get initialized when things load, right?

Reply:
This is before instantiating `Modernizer`. We are reading the strings from the files and adding to these lists. Then, we are passing this to create a `Modernizer` object. In the constructor, `Utils.createImmutableSet` is being used. 

https://github.com/HubSpot/modernizer-maven-plugin/blob/method-level-suppressions/modernizer-maven-plugin/src/main/java/org/gaul/modernizer_maven_plugin/Modernizer.java#L78

Reply:
Also, I think it should be changes to a `Set` instead of a `List`.